flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -680,10 +680,20 @@ private static void validateInputType(Type t, TypeInformation<?> inType) {
 		}
 	}
 	
-	private static void validateInputType(Class<?> baseClass, Class<?> clazz, int inputParamPos, TypeInformation<?> inType) {
+	private static void validateInputType(Class<?> baseClass, Class<?> clazz, int inputParamPos, TypeInformation<?> inTypeInfo) {
 		ArrayList<Type> typeHierarchy = new ArrayList<Type>();
+
+		// try to get generic parameter
+		Type inType;
+		try {
+			inType = getParameterType(baseClass, typeHierarchy, clazz, inputParamPos);
+		}
+		catch (Exception e) {
+			return; // skip input validation e.g. for raw types
```

Comment:
Actually yes, the input validation is only a "nice to have" feature and helps the "normal" user. If the getParameterType() does not work, the validation should be skipped. We can also change it to "InvalidTypeException" if you want.